### PR TITLE
Adding support for Google Cloud Functions resource

### PR DESCRIPTION
**Pull request by ekifel**
Issue #311 

```
NAME                                          MONTHLY QTY  UNIT         PRICE   HOURLY COST  MONTHLY COST 
google_cloudfunctions_function.my_function                                                                 
├─ CPU                                            600,000  GHz-seconds   1e-05       0.0082        6.0000  
├─ Memory                                         375,000  GB-seconds    3e-06       0.0013        0.9375  
├─ Invocations                                 10,000,000  invocations   4e-07       0.0055        4.0000  
└─ Outbound data transfer                             100  GB           0.1200       0.0164       12.0000  
Total                                                                                0.0314       22.9375  
```

### CODE REVIEW

File: internal/providers/terraform/google/cloudfunctions_function.go [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+					AttributeFilters: []*schema.AttributeFilter{
+						{Key: "description", Value: strPtr("Invocations")},
+					},
+				},
+				PriceFilter: &schema.PriceFilter{
+					StartUsageAmount: strPtr("2000000"), // use the non-free tier
+				},
+			},
+			{
+				Name:            "Outbound Data",
```
Comment (alikhajeh1) [inline internal/providers/terraform/google/cloudfunctions_function.go:90]:
```suggestion
				Name:            "Outbound data transfer",
```

File: internal/providers/terraform/google/cloudfunctions_function.go [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+				Unit:            "GB",
+				UnitMultiplier:  1,
+				MonthlyQuantity: networkEgrees,
+				ProductFilter: &schema.ProductFilter{
+					VendorName:    strPtr("gcp"),
+					Region:        strPtr("us-east1"),
```
Comment (alikhajeh1) [inline internal/providers/terraform/google/cloudfunctions_function.go:96]:
```suggestion
					Region:        strPtr(region),
```
?

File: internal/providers/terraform/google/cloudfunctions_function_test.go [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+import (
+	"testing"
+
+	"github.com/infracost/infracost/internal/providers/terraform/tftest"
+	"github.com/infracost/infracost/internal/schema"
+	"github.com/infracost/infracost/internal/testutil"
+)
+
+func TestCloudFunctions(t *testing.T) {
```
Comment (alikhajeh1) [inline internal/providers/terraform/google/cloudfunctions_function_test.go:11]: Seems like this needs a _usage test too

**Review by alikhajeh1 — CHANGES_REQUESTED** [review]
Awesome! I left my suggestions

**Comment by ekifel** [comment]
@alikhajeh1 this is ready for review/merge

File: infracost-usage-example.yml [hunk 4]
```diff
@@ -161,6 +161,11 @@ resource_usage:
   #
   # Terraform GCP resources
   #
+  google_cloudfunctions_function.my_function:
+    request_duration_ms: 300 # Average duration of each request in milliseconds.
+    monthly_function_invocations: 10000000 # Monthly number of function invocations.
+    monthly_outbound_data_gb: 100 # Monthly data transferred from the function out to somewhere else in GB.
```
Comment (alikhajeh1) [inline infracost-usage-example.yml:167]:
```suggestion
    request_duration_ms: 300               # Average duration of each request in milliseconds.
    monthly_function_invocations: 10000000 # Monthly number of function invocations.
    monthly_outbound_data_gb: 100          # Monthly data transferred from the function out to somewhere else in GB.
```

File: internal/providers/terraform/google/cloudfunctions_function.go [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+	if u != nil && u.Get("monthly_function_invocations").Exists() {
+		invocations = decimal.NewFromInt(u.Get("monthly_function_invocations").Int())
+	}
+
+	var networkEgrees *decimal.Decimal
+	if u != nil && u.Get("monthly_outbound_data_gb").Exists() {
+		networkEgrees = decimalPtr(decimal.NewFromInt(u.Get("monthly_outbound_data_gb").Int()))
+	}
+
+	requestDuration := decimal.Zero
```
Comment (alikhajeh1) [inline internal/providers/terraform/google/cloudfunctions_function.go:44]: The pricing page mentions "compute time is measured in 100ms increments, rounded up to the nearest increment" So I think we should:
1. Default this to 100
2. Inside the if-statement, round **up** the user input to the nearest 100ms, e.g. if they provide 240, we set it to 300.

File: internal/providers/terraform/google/cloudfunctions_function.go [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+		256:  decimal.NewFromInt(400),
+		512:  decimal.NewFromInt(800),
+		1024: decimal.NewFromInt(1400),
+		2048: decimal.NewFromInt(2400),
+		4096: decimal.NewFromInt(4800),
+	}
+
+	cpuSize := cpuMapping[int(memorySize.IntPart())]
+
+	invocations := decimal.Zero
```
Comment (alikhajeh1) [inline internal/providers/terraform/google/cloudfunctions_function.go:34]: This should default to `*decimal.Decimal` as we don't know the number

**Review by alikhajeh1 — CHANGES_REQUESTED** [review]
Looking good! Two minor changes, sorry I should have spotted those from the pricing page earlier.

**Comment by alikhajeh1** [comment]
Plz see https://github.com/infracost/infracost/pull/399/commits/23fde606bd1c16e4a4af20fe3be8c607dcb828fa for my changes

**Pull request merged**